### DISCUSSION TITLE
Having trouble with folds and session restore

### DISCUSSION THREAD

**Discussion by adrian5** [olimorris/persisted.nvim#88] [General]
I don't know if this is an issue with persisted.nvim or Neovim itself.

When loading a session, nvim experiences error `E490: No fold found`. This is apparently a [telescope issue](https://github.com/nvim-telescope/telescope.nvim/issues/699) as well.

I can give up on `folds` from `sessionoptions` to prevent this error from affecting startup (session restore). Either way, folds won't work in the session-restored buffers unless I have nvim recompute them:

```lua
vim.api.nvim_create_autocmd("User", {
  desc = "Recompute folds after session is loaded",
  pattern = "PersistedLoadPost",
  group = vim.api.nvim_create_augroup("PersistedHooks", {}),
  callback = function()
    vim.defer_fn(function()
      vim.cmd("normal! zx")
    end, 50) -- Small delay seems necessary
  end
})
```

Do you experience this as well when restoring buffers and autoloading? I.e. `za` throws `E490` unless you invoke `zx` first, or open another buffer.

If so, I wonder if you could add a workaround to the plugin itself. Maybe it misses some nvim default action when restoring buffers, the same way telescope does.

**Comment by olimorris** [comment]
I don't recall ever coming across that error code. Are you on Neovim 0.9.2?

My sessionoptions are set in my config, [here](https://github.com/olimorris/dotfiles/blob/ec746e5c956043dde055871008a4185250cbe2bb/.config/nvim/lua/config/options.lua#L60). I am using [nvim-ufo](https://github.com/kevinhwang91/nvim-ufo) these days too. Might be worth playing around with that

**Reply by adrian5** [reply]
I couldn't nail it down, it's not your plugin and happens with plain `:mksession` an `:source` as well. I'm on Neovim 0.9.2.

I've now set up nvim-ufo as I simply don't have the time to dig any deeper; ufo seems to handle folds in sessions just fine. Maybe the problem will resolve itself once **telescope** and **nvim-treesitter** tackle (seemingly) related issues.

Thanks!

**Reply by olimorris** [reply]
The joys of Neovim. Glad you got a resolution. My latest pain is Neovim seemingly exiting after holding down `j` too many times 😂
